src/DynamoCore/Search/NodeSearchModel.cs

```diff
@@ -237,14 +237,7 @@ internal IEnumerable<NodeSearchElement> Search(string search, LuceneSearchUtilit
             if (luceneSearchUtility != null)
             {
                 //The DirectoryReader and IndexSearcher have to be assigned after commiting indexing changes and before executing the Searcher.Search() method, otherwise new indexed info won't be reflected
-                if (luceneSearchUtility.writer != null)
-                {
-                    luceneSearchUtility.dirReader = luceneSearchUtility.writer.GetReader(applyAllDeletes: true);
-                }
-                else
-                {
-                    luceneSearchUtility.dirReader = DirectoryReader.Open(luceneSearchUtility.indexDir);
-                }
+                luceneSearchUtility.dirReader = luceneSearchUtility.writer != null ? luceneSearchUtility.writer.GetReader(applyAllDeletes: true) : DirectoryReader.Open(luceneSearchUtility.indexDir);
```

Comment:
I am considering adding getter for `dirReader` and `Searcher` because they are always the same code so we dont have to leak these Lucence specific code everywhere in Dynamo code